### ISSUE TITLE
isolatedDeclarations doesn't support `tsconfig` options

### ISSUE THREAD

**Issue by tombl** [oxc-project/oxc#3958]
Some `tsconfig` options are relevant for declaration emit, but there's currently no way to pass these.

```js
> code = "/** @internal */ export function hello() {}"

> require("oxc-transform").isolatedDeclaration("example.ts", code)
{ errors: ['TS9007: Function must have an explicit return type annotation with --isolatedDeclarations.'] }

> require("typescript").transpileDeclaration(code, {})
{
  diagnostics: [{
    messageText: 'Function must have an explicit return type annotation with --isolatedDeclarations.',
  }]
}

> require("typescript").transpileDeclaration(code, { compilerOptions: { stripInternal: true } })
{ outputText: 'export {};\n' }
```

**Comment by Dunqing** [comment]
Thanks for trying it out, in addition to supporting passing `tsconfig` options we also need to check the `jsdoc` @Boshen cc

**Comment by Boshen** [comment]
https://x.com/TitianCernicova/status/1808132493469442398

> If stripInternal is specified TSC will respect the internal JSDoc tag even if ID is specified, and this is the behavior that is expected of other DTS implementations.

> internal is useful as an escape hatch for functions you might export from a module, but don't actually want to export from your package so they don't need to make it into declaration files.

**Comment by Boshen** [comment]
Close as not planned. It seems like ID shouldn't check jsdoc nor tsconfig :-/

**Comment by tombl** [comment]
It's my understanding that isolatedDeclarations is mostly about not running a full typechecker to emit declarations.
Are there potential perf issues with supporting this or is it simply out of scope?

**Comment by Dunqing** [comment]
I've found that most of the popular libraries use `@internal` directive and strip them by `stripInternal`. So I think we should consider supporting it.
